lib/utils/deprecate.js

```diff
@@ -0,0 +1,120 @@
+var util = require('util');
+var _ = require('lodash');
+var stackTrace = require('stack-trace');
+var colorize = require('../ui/colorize');
+var Logger = require('../logger');
+
+var logger = new Logger();
+
+var delayedMessages = [];
+var isInitialized = false;
+
+exports.initialize = function (mustShowWarnings) {
+    isInitialized = true;
+
+    if (mustShowWarnings) {
+        logger.showWarnings();
+    } else {
+        logger.hideWarnings();
+    }
+
+    printDelayedMessages();
+};
+
+exports.deprecate = function (message) {
+    var error = checkMessage(message);
+
+    if (error) {
+        throw error;
+    }
+
+    if (!isInitialized) {
+        delayedMessages.push(message);
+    } else {
+        printMessage(message);
+    }
+};
+
+function checkMessage(message) {
+    if (typeof message.module !== 'string') {
+        return new Error('Please provide deprecated module name in message.module');
+    }
+
+    if (typeof message.since !== 'string') {
+        return new Error('Please provide remove version in message.since');
+    }
+
+    return null;
```

Comment:
```js
['module', 'since'].forEach(function(field) {
    if(!message[field]) {
        throw new Error('Missing required field ' + field);
    }
});
```